frontend/src/v5/ui/components/viewer/drawingViewer/drawingViewer.component.tsx

```diff
@@ -0,0 +1,101 @@
+/**
+ *  Copyright (C) 2024 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { formatMessage } from '@/v5/services/intl';
+import { ToolbarButton } from '@/v5/ui/routes/viewer/toolbar/buttons/toolbarButton.component';
+import { ToolbarContainer, MainToolbar } from '@/v5/ui/routes/viewer/toolbar/toolbar.styles';
+import { useRef, useState } from 'react';
+import ZoomOutIcon from '@assets/icons/viewer/zoom_out.svg';
+import ZoomInIcon from '@assets/icons/viewer/zoom_in.svg';
+
+import { FileInputField } from '@controls/fileInputField/fileInputField.component';
+import { Button } from '@controls/button/button.component';
+import { FormattedMessage } from 'react-intl';
+import { SvgViewer } from './svgViewer.component';
+import { PanZoomHandler, centredPanZoom } from './panzoom/centredPanZoom';
+
+
+
+export const DrawingViewer = () => {
+	const [svgContent, setSvgContent] = useState('');
+	const [imgContent, setImgContent] = useState('');
+	const [zoomHandler, setZoomHandler] = useState<PanZoomHandler>();
+
+	const imgRef = useRef<HTMLImageElement | SVGSVGElement>();
+
+	const onClickButton = async ([file]: File[]) => {
+		if (file.type.includes('svg')) { 
+			setSvgContent(await file.text());
+			setImgContent('');
+		}
+
+		if (file.type.includes('png')) { 
+			var base64String = btoa(String.fromCharCode.apply(null, new Uint8Array(await file.arrayBuffer())));
```

Comment:
I'm getting this error when attempting to upload certain PNGs.
![image](https://github.com/3drepo/3drepo.io/assets/62481250/f209d3f9-78b3-4a51-95ac-c4edba86b567)
It seems like I can only upload very smalls PNGs. This one was only 272KB
![image](https://github.com/3drepo/3drepo.io/assets/62481250/e7b12278-7cb3-4e6b-9d65-f925f767993e)

Reply:
It doesnt matter. the selection of pngs here and convertion is just for  you to pick a png from your computer. When this is actually working it will get the url and not convert anything.